### PR TITLE
Spec for Array#any?

### PR DESCRIPTION
**Pull request by maxfurman**
Low hanging fruit, I know

### CODE REVIEW

File: core/array/any_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+describe "Array#any?" do
+  describe 'with no block given' do 
+    it "checks if there are any members of an array" do
```
Comment (eregon) [inline core/array/any_spec.rb]: I didn't catch it for Hash#any?, but it is actually the same (could you fix in a separate commit?):
`any?` without block behaves as if you passed `{ |e| e }` so for instance `[nil].any?` is `false`.

Comment (eregon) [inline core/array/any_spec.rb]: Could you adapt the spec description to imply this and add an example?

Comment (maxfurman) [inline core/array/any_spec.rb]: Sure. Although, I am having trouble constructing a hash with falsy values such that `any?` is false.

`{nil => nil}.any? #=> true`
`{false => false}.any? #=> true`
`{nil => false}.any? #=> true`

Comment (eregon) [inline core/array/any_spec.rb]: Ah good find! Indeed for Hash#any? this detail does not matter since an Array is yielded and that is always truthy.

**Comment by eregon** [comment]
Very nice, merged!
Thanks for this :smiley:

**Pull request merged**